docs/src/snap/howto/epa.md

```diff
@@ -0,0 +1,1061 @@
+properly. Then you can create a Pod that explicitly requests one 1G Huge Page
+in its resource limits:
+
+```
+cat <<EOF | sudo k8s kubectl apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: hugepage-test-ubuntu
+spec:
+  containers:
+  - name: ubuntu-container
+    image: ubuntu:latest
+    command: ["sleep", "infinity"]
+    resources:
+      limits:
+        hugepages-1Gi: 1Gi
+        memory: 128Mi
+      requests:
+        memory: 128Mi
+EOF
+```
+
+Note: To ensure proper resource management and prevent conflicts, Kubernetes
+enforces that a pod requesting HugePages also explicitly requests a minimum
+amount of either CPU or standard memory resources.
+
+Now, let’s check if the 1 Gi HugePage was allocated in the pod:
+
+```
+sudo k8s kubectl describe pod hugepage-test-ubuntu
+``` 
+
+This should produce output indicating the request:
+
+```
+....
+    Limits:
+      hugepages-1Gi:  1Gi
+      memory:         128Mi
+    Requests:
+      hugepages-1Gi:  1Gi
+      memory:         128Mi
+....
+
+```
+
+## Testthe real-time kernel
+
+First, verify that real-time kernel is enabled in the worker node by checking
+if “PREEMPT RT” appears after running the `uname -a` command:
+
+```
+uname -a
+```
+
+The output should show the “PREEMPT RT” identifier:
+
+```
+Linux pc6b-rb4-n3 6.8.1-1004-realtime #4~22.04.1-Ubuntu SMP PREEMPT_RT Mon Jun 24 16:45:51 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
+```
+
+The test will use cyclictest, commonly used to assess the real-time performance
+of a system, especially when running a real-time kernel. It measures the time
+it takes for a thread to cycle between high and low priority states, giving you
+an indication of the system's responsiveness to real-time events.  Lower
+latencies typically indicate better real-time performance.
+
+The output of cyclictest will provide statistics like:
+
+* Average latency: The average time taken for a cycle.  
+* Minimum latency: The shortest observed cycle time.  
+* Maximum latency: The longest observed cycle time.
+
+Create a pod that will run cyclictest tool with specific options:
+
+- `-l 1000000`: Sets the number of test iterations to 1 million.  
+- `-m`: Measures the maximum latency.  
+- `-p 80`: Sets the real-time scheduling priority to 80 (a high priority, typically used for real-time tasks).  
+- `-t 1`: Specifies CPU core 1 to be used for the test.
+
+```
+cat <<EOF | sudo k8s kubectl apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: realtime-kernel-test
+spec:
+  containers:
+  - name: realtime-test
+    image: ubuntu:latest
+    command: ["/bin/bash"]
+    args: ["-c", "apt-get update && apt-get install rt-tests -y && cyclictest -l 1000000 -m -p 80 -t 1"]
+    securityContext:
+      privileged: true     
+EOF
+```
+
+Get the pod logs to verify that the test is running:
+
+```
+sudo k8s kubectl logs realtime-kernel-test -f
+
+```
+
+```
+...
+# /dev/cpu_dma_latency set to 0us
```

Comment:
I'm seeing this in the sidebar
![image](https://github.com/user-attachments/assets/0b88c610-a3c9-4050-91c0-3198c0313cd8)